File: src/main/java/com/itsci/mju/maebanjumpen/hire/dto/HireDTO.java
package com.itsci.mju.maebanjumpen.hire.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties; // 💡 เพิ่ม import นี้
import com.itsci.mju.maebanjumpen.review.dto.ReviewDTO;
import com.itsci.mju.maebanjumpen.partyrole.dto.HirerDTO;
import com.itsci.mju.maebanjumpen.partyrole.dto.HousekeeperDTO;
import com.itsci.mju.maebanjumpen.skilltype.dto.SkillTypeDTO;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;
import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true) // 🎯 แก้ไข: สั่งให้ Jackson เพิกเฉยต่อ field ที่ไม่รู้จัก (เช่น "report")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class HireDTO {
	private Integer hireId;
	private String hireName;
	private String hireDetail;
	private Double paymentAmount;
	private LocalDateTime hireDate;
	private LocalDate startDate;
	private LocalTime startTime;
	private LocalTime endTime;
	private String location;
	private String jobStatus;
	private List<String> progressionImageUrls;

	// Use simple DTOs for relationships
	private HirerDTO hirer;
	private HousekeeperDTO housekeeper;
	private SkillTypeDTO skillType;
	private ReviewDTO review; // Use ReviewDTO for OneToOne

}

File: src/main/kotlin/com/itsci/mju/maebanjumpen/hire/dto/HireDTO.kt
package com.itsci.mju.maebanjumpen.hire.dto

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.itsci.mju.maebanjumpen.partyrole.dto.HirerDTO
import com.itsci.mju.maebanjumpen.partyrole.dto.HousekeeperDTO
import com.itsci.mju.maebanjumpen.review.dto.ReviewDTO
import com.itsci.mju.maebanjumpen.skilltype.dto.SkillTypeDTO
import java.time.LocalDate
import java.time.LocalDateTime
import java.time.LocalTime

@JsonIgnoreProperties(ignoreUnknown = true)
data class HireDTO(
    var id: Long? = 0,
    var hireName: String? = null,
    var hireDetail: String? = null,
    var paymentAmount: Double? = null,
    var hireDate: LocalDateTime? = null,
    var startDate: LocalDate? = null,
    var startTime: LocalTime? = null,
    var endTime: LocalTime? = null,
    var location: String? = null,
    var jobStatus: String? = null,
    var progressionImageUrls: List<String>? = null,
    var hirer: HirerDTO? = null,
    var housekeeper: HousekeeperDTO? = null,
    var skillType: SkillTypeDTO? = null,
    var review: ReviewDTO? = null
)

